config/default.yml

```diff
@@ -394,6 +394,13 @@ RSpec/MultipleExpectations:
   VersionChanged: '1.21'
   StyleGuide: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/MultipleExpectations
 
+RSpec/MultipleMemoizedHelpers:
+  Description: Checks if example groups contain too many `let` and `subject` calls.
+  Enabled: false
```

Comment:
I would personally be happy to enable it by default with `AllowSubject: true` and `Max: 5`.

Reply:
Did you run stats on some real world projects to see how many offenses are generated at different `Max` levels?

Reply:
Also we have a cop for multiple subjects, so I doubt the need to check subject calls at all

Reply:
I'm thinking of gathering a Real World RSpec, since Real World Ruby/Real World Rails don't use RSpec much.

Reply:
Here it goes https://github.com/pirj/real-world-rspec/

Had to remove all `.rubocop*.yml` files around those projects before running `rubocop` on them. RuboCop hiccups when there's an incompatible `TargetRubyVersion` or a renamed cop in the configuraion.

```
$ rubocop --only RSpec/MultipleMemoizedHelpers ~/source/real-world-rspec
Inspecting 47132 files
```
Max helpers | Offence count
--- | ---
`Max: 5` | 24640
`Max: 7` | 14418
`Max: 10` | 6093

Notable examples (@mockdeep, fasten your seatbelts):
10 lets https://github.com/forem/forem/blob/master/spec/support/api_analytics_shared_examples.rb#L2
20 lets https://github.com/chef/chef/blob/master/spec/unit/provider/systemd_unit_spec.rb#L23
30 lets https://github.com/chef/chef/blob/master/spec/unit/policy_builder/policyfile_spec.rb#L24
40 lets https://github.com/diaspora/diaspora/blob/develop/spec/integration/migration_service_spec.rb#L7
50 lets https://github.com/chef/chef/blob/master/spec/functional/resource/dsc_script_spec.rb#L71

I can prepare the report per-repo, but I guess those two, diaspora and chef, are the worst offenders.

Reply:
> Also we have a cop for multiple subjects, so I doubt the need to check subject calls at all

I think `subject` counts towards memoized helpers, as its differences from `let` are subtle.

Even though with the default configuration we only allow one `subject`, some may want zero of them. I'm quite ok to make `AllowSubject: true` the default though for those who don't.

Reply:
The numbers should be slightly lower now when `subject` doesn't count towards the total number by default.

A [bug in RuboCop](https://github.com/rubocop-hq/rubocop/issues/8465) prevents from building a script that would run `rubocop` on each of 35 repositories and tell which ones offend the new cops most of all.

Here's an [example pull request](https://github.com/rubocop-hq/rubocop/pull/8447) showing that it's possible to reduce the use memoized helpers to a reasonable number.